src/compute-client/src/controller/replica.rs

```diff
@@ -128,14 +139,38 @@ struct ReplicaTask<T> {
     /// A channel upon which commands intended for the replica are delivered.
     command_rx: UnboundedReceiver<ComputeCommand<T>>,
     /// A channel upon which responses from the replica are delivered.
-    response_tx: UnboundedSender<ComputeResponse<T>>,
+    response_tx: UnboundedSender<ReplicaResponse<T>>,
     /// Orchestrator responsible for setting up computeds
     orchestrator: ComputeOrchestrator,
     /// A number (technically, pair of numbers) identifying this incarnation of the replica.
     /// The semantics of this don't matter, except that it must strictly increase.
     epoch: ComputeStartupEpoch,
 }
 
+fn metrics_stream(
+    orchestrator: ComputeOrchestrator,
+    instance_id: ComputeInstanceId,
+    replica_id: ReplicaId,
+) -> BoxStream<'static, Result<Vec<ServiceProcessMetrics>, anyhow::Error>> {
+    const METRICS_INTERVAL: Duration = Duration::from_secs(10);
+
+    // TODO[btv] -- I tried implementing a `watch_metrics` function,
+    // similar to `watch_services`, but it crashed due to
+    // https://github.com/kube-rs/kube/issues/1092 .
+    //
+    // If `metrics-server` can be made to fill in `resourceVersion`,
+    // or if that bug is fixed, we can try that again rather than using this inelegant
+    // loop.
+    let s = async_stream::stream! {
+        let mut interval = tokio::time::interval(METRICS_INTERVAL);
+        loop {
+            interval.tick().await;
+            yield orchestrator.fetch_replica_metrics(instance_id, replica_id).await;
+        }
+    };
+    s.boxed()
+}
```

Comment:
Would it be easier to just add `interval.tick()` to the `select!` in line  275 and then call `fetch_replica_metrics` ?

Reply:
With that design, if `fetch_replica_metrics` is slow for some reason, `run_message_loop` might wait for a long time for it to return, and therefore delay handling commands/responses (which are more important).

In the current design, we don't enter the handler for `metrics_result` until a result is actually ready, and the handler is non-blocking.

Reply:
I see, right.